automata/regex/parser.py

```diff
@@ -140,9 +140,9 @@ def intersection(self, other):
             if epsilon_transitions_b is not None:
                 state_dict = new_transitions.setdefault(curr_state_name, {})
                 state_dict.setdefault('', set()).update(
-                    map(get_state_name, product([q_a], epsilon_transitions_b))
+                    map(get_state_name, zip(repeat(q_a), epsilon_transitions_b))
```

Comment:
@eliotwrobson What's the advantage of using `repeat(q_a)` over `[q_a]`?

Reply:
So I saw someone do this in an itertools video and mainly wanted to try it out, but after some timing turns out its slower... Another lesson on why it's not good to try and be too clever 😅

Reply:
@eliotwrobson Well, how many iterations are those benchmarks based off of? I am curious if one solution scales better than the other.

Reply:
This was 1 million. I'm actually curious now, let me try some different numbers.

Reply:
@eliotwrobson Just checking on the status of your benchmarking, since this thread is the last unresolved conversation on this PR.

Reply:
The changes discussed in this thread actually got merged in #130, but here is more detailed benchmarking:

![image](https://user-images.githubusercontent.com/1345068/228988265-f0b29e84-15b5-4c54-96c1-9faff7eb580a.png)

Even at the smaller sizes, the difference is somewhat noticeable.

Reply:
@eliotwrobson Oh okay great! I'm satisfied then lol. Will resolve this thread and review the rest of your PR in preparation to approve/merge.